src/lib.rs

```diff
@@ -495,6 +495,43 @@ impl Client {
         nonce: Option<String>,
         signer: &impl Signer,
         url: &str,
+    ) -> Result<BytesResponse, Error> {
+        let mut retries = 3;
+
+        let mut nonce = nonce;
+        loop {
+            let mut response = self.inner_post(payload, nonce.clone(), signer, url).await?;
+            if response.parts.status == StatusCode::BAD_REQUEST {
+                let body = response.body.into_bytes().await.map_err(Error::Other)?;
+                let problem: Problem = serde_json::from_slice(&body)?;
```

Comment:
Nit: I prefer to avoid type annotation, in favor of turbofishing where necessary.